x-pack/plugin/rank-vectors/src/main/java/org/elasticsearch/xpack/rank/vectors/mapper/RankVectorsFieldMapper.java

```diff
@@ -180,7 +180,17 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             return new ArraySourceValueFetcher(name(), context) {
                 @Override
                 protected Object parseSourceValue(Object value) {
-                    return value;
+                    List<?> outerList = (List<?>) value;
+                    List<float[]> vectors = new ArrayList<>(outerList.size());
+                    for (Object o : outerList) {
+                        List<?> innerList = (List<?>) o;
+                        float[] vector = new float[innerList.size()];
+                        for (int i = 0; i < vector.length; i++) {
+                            vector[i] = ((Number) innerList.get(i)).floatValue();
+                        }
+                        vectors.add(vector);
+                    }
+                    return vectors;
```

Comment:
```suggestion
                    List<?> outerList = (List<?>) value;
                    List<Object> vectors = new ArrayList<>(outerList.size());
                    for (Object o : outerList) {
                        if (o instanceof List) {
                            List<?> innerList = (List<?>) o;
                            float[] vector = new float[innerList.size()];
                            for (int i = 0; i < vector.length; i++) {
                                vector[i] = ((Number) innerList.get(i)).floatValue();
                            }
                            vectors.add(vector);
                        } else {
                            vectors.add(o);
                        }
                    }
```

Something like this, as we allow hex strings.